Bottom: 0ca8708caa50faeac8b90d39f9210e6ea930457f
Top:    c69a9f917b453c62496e7544d1136f31ba87d14e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-15 19:49:20 +0000

Refresh of drop-in-pcb-gl-code-various-mess

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 93eb2fb..3e41c35 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -462,6 +462,12 @@ ghid_invalidate_current_gc (void)
 static int
 use_gc (hidGC gc)
 {
+  if (gc->me_pointer != &ghid_hid)
+    {
+      fprintf (stderr, "Fatal: GC from another HID passed to GTK HID\n");
+      abort ();
+    }
+
   if (current_gc == gc)
     return 1;
